Dashboard: add a view that shows tests in order of number of failures

azharcodeit opened the pull request:
**Context:**
 
This bug is about showing a simple view of the tests that flake the most. We don't need to show all tests, but maybe it would be good to show the top 100 tests and the number of times they had unexpected results.

**Solution:**

- Created an app route /most-flaky for a view, which contains a table that shows Test Path and Number of Unexpected Results
- Added a new function called `most_flaky` to retrieve the flakiest tests from the database. This function calculates the total unexpected count for each test and sorts them in descending order. 
- Added a limit parameter as a query parameter of the request



<br class="Apple-interchange-newline"> 


- [x] This PR fixes #15 
- [x] `python3 -m intermittent_tracker.tests` doesn't report any error

### Test plan
- Successful test run:
`python3 -m intermittent_tracker.tests` doesn't report any error`
- Demo of a newly created view:



https://github.com/servo/intermittent-tracker/assets/31756707/47c8318a-1229-4e23-bfa4-ed0fc7093031

azharcodeit committed:
add: added a view that shows tests in order of number of failures

azharcodeit committed:
refactor: changed the query to work with tests table only, by loading path & unexpected count

azharcodeit committed:
refactor: changed the query to work with tests table only, by loading path & unexpected count

mrobinson commented on intermittent_tracker/dashboard.py:
```diff
@@ -44,3 +44,18 @@ def query(request):
             result['unknown'].append(test)
     return json.dumps(result)
+
+def flaky_tests(limit=100):
```
Can this be called something like `most_flaky`?

mrobinson commented on intermittent_tracker/flask_server.py:
```diff
@@ -54,6 +54,11 @@ def dashboard_query():
 def index():
     return send_from_directory('static', 'index.html')
 
+@app.route('/flaky_tests')
```
I recommend `most-flaky` here as well.

mrobinson commented on intermittent_tracker/flask_server.py:
```diff
@@ -54,6 +54,11 @@ def dashboard_query():
     return send_from_directory('static', 'index.html')
 
+@app.route('/flaky_tests')
+def flaky_tests():
```
It's great that you made the method in `dashboard.py` take a parameter for number of tests, but can you also allow passing a URL parameter here to control that as well?

mrobinson commented on intermittent_tracker/templates/flaky_tests.html:
```diff
@@ -0,0 +1,30 @@
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Top 100 Flaky Tests</title>
```
This title would have to change to be a bit more generic. Perhaps it could be "Flakiest Tests".

mrobinson reviewed (changes requested):
Nice work. Please also test this with the data I uploaded to the original issue to make sure that performance and behavior are as you expected them.

azharcodeit committed:
add: added limit parameter as a query parameter of the request

azharcodeit commented:
Hey @mrobinson, thanks for reviewing my PR!

I've addressed the comments that you left. Also, I've updated the summary of this PR, which now includes new description and demo video using data that you provided in the issue.

Could you please take a look at it one more time?

azharcodeit committed:
refactor: remove unused import

mrobinson reviewed (approved):
Nicely done!
